Pomodoro/app/src/main/java/no/uia/ikt205/pomodoro/MainActivity.kt

```diff
@@ -1,52 +1,90 @@
 import android.os.Bundle
 import android.os.CountDownTimer
+import android.view.Gravity
 import android.view.View
-import android.widget.Button
-import android.widget.TextView
-import android.widget.Toast
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
 import no.uia.ikt205.pomodoro.util.millisecondsToDescriptiveTime
 
-class MainActivity : AppCompatActivity() {
+
+class MainActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener {
 
     lateinit var timer:CountDownTimer
     lateinit var startButton:Button
     lateinit var coutdownDisplay:TextView
+    lateinit var dropDown:Spinner
+
+    private var hasTimerStarted: Boolean = false
 
-    val timeToCountDownInMs = 5000L
-    val timeTicks = 1000L
+    private val countdownTimeArray = arrayOf<Long>(
+        1800000L,
+        3600000L,
+        5400000L,
+        7200000L
+    )
+    private var timeToCountDownInMs = countdownTimeArray[0]
+    private val timeTicks = 1000L
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-       startButton = findViewById<Button>(R.id.startCountdownButton)
-       startButton.setOnClickListener(){
-           startCountDown(it)
-       }
-       coutdownDisplay = findViewById<TextView>(R.id.countDownView)
+        dropDown = findViewById<View>(R.id.dropdown_menuTimes) as Spinner
+        val aa = ArrayAdapter(this, android.R.layout.simple_spinner_item, countdownTimeArray)
+        aa.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
 
+        with(dropDown) {
+            adapter = aa
+            setSelection(0, false)
+            onItemSelectedListener = this@MainActivity
+            prompt = "Select time interval"
+            gravity = Gravity.CENTER
+        }
+        startButton = findViewById<Button>(R.id.startCountdownButton)
+        startButton.setOnClickListener(){
+            startCountDown(it)
+        }
+        coutdownDisplay = findViewById<TextView>(R.id.countDownView)
     }
 
-    fun startCountDown(v: View){
+    private fun startCountDown(v: View){
+        if (hasTimerStarted) {
+            Toast.makeText(this@MainActivity, "Timer is already running", Toast.LENGTH_SHORT).show()
+            return
+        }
 
-        timer = object : CountDownTimer(timeToCountDownInMs,timeTicks) {
+        timer = object : CountDownTimer(timeToCountDownInMs, timeTicks) {
             override fun onFinish() {
-                Toast.makeText(this@MainActivity,"Arbeidsøkt er ferdig", Toast.LENGTH_SHORT).show()
+                Toast.makeText(this@MainActivity, "Arbeidsøkt er ferdig", Toast.LENGTH_SHORT).show()
+                hasTimerStarted = false
             }
 
             override fun onTick(millisUntilFinished: Long) {
                updateCountDownDisplay(millisUntilFinished)
             }
         }
-
         timer.start()
+        hasTimerStarted = true
```

Comment:
Ser at du bruker variabelen hasTimerStarted videre i koden for å løse nedtellings-bugen. Bra jobba!